### PR TITLE
feat(Masuru): add presence

### PR DESCRIPTION
**Pull request by aomkoyo**
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
image
image
image
image
image

</details>

### CODE REVIEW

**Comment by aomkoyo** [comment]
image
image
image
image
image

File: websites/M/Masuru/metadata.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		"en": "Content for everyone to enjoy with us."
+	},
+	"url": [
+		"www.masuru.in.th",
+		"masuru.in.th",
+		"cloud.masuru.in.th",
+		"forum.masuru.in.th"
+	],
+	"version": "1.0.0",
+	"logo": "https://cdn.masuru.in.th/assets/masuru/apple-touch-icon.webp",
```
Comment (theusaf) [inline websites/M/Masuru/metadata.json:18]: Logos should be 512x512px

File: websites/M/Masuru/presence.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
+const presence = new Presence({
+		clientId: "861582205028794418",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets { // Other default assets can be found at index.d.ts
+	Logo = "masuru",
+	Bot = "logo",
+}
```
Comment (theusaf) [inline websites/M/Masuru/presence.ts:9]: Use image urls instead of discord asset names

File: websites/M/Masuru/presence.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+			if (title) {
+				presenceData.details = title;
+				(presenceData.largeImageKey = Assets.Logo),
+					// Add Button View Page
+					(presenceData.buttons = [
+						{
+							label: "View Page",
+							url: href,
+						},
+					]);
```
Comment (theusaf) [inline websites/M/Masuru/presence.ts:28]: Separate these assignments with semicolons rather than commas

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/M/Masuru/presence.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+						presenceData.details = `Status - ${title}`;
+						break;
+					default:
+						if (pathname_.startsWith("/dashboard")) {
+							presenceData.details = `Dashboard - ${title}`;
+							presenceData.state = pathname_.split("/").reverse()[0];
+							if (presenceData.state === "discovery") {
+								presenceData.buttons = [
+									{
+										label: "Discovery Page",
```
Comment (theusaf) [inline websites/M/Masuru/presence.ts:52]: Button label should be a verb.
- View Discovery
- View Discovery Page
- Discover
- etc.

File: websites/M/Masuru/presence.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+			const title = document.querySelector("title").textContent;
+			if (title) {
+				presenceData.details = title;
+				presenceData.largeImageKey = Assets.Logo;
+				presenceData.buttons = [
+					{
+						label: "View Page",
+						url: href,
+					},
+				];
```
Comment (theusaf) [inline websites/M/Masuru/presence.ts:27]: This button is generic and may redirect to the home page, which is not allowed according to current guidelines.
```suggestion
```

**Review by theusaf — CHANGES_REQUESTED** [review]

**Comment by tag27** [comment]
Why is the name of the presence Lilia and not the name of the website (Masaru)?

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**